nova_vm/src/ecmascript/abstract_operations/testing_and_comparison.rs

```diff
@@ -53,37 +56,61 @@ pub(crate) fn is_array(_agent: &Agent, argument: Value) -> JsResult<bool> {
 /// The abstract operation IsCallable takes argument argument (an ECMAScript
 /// language value) and returns a Boolean. It determines if argument is a
 /// callable function with a [[Call]] internal method.
-pub(crate) fn is_callable(argument: impl IntoValue) -> bool {
+///
+/// > #### Note
+/// > Nova breaks with the specification to narrow the types automatically, and
+/// > returns an `Option<Function>`. Eventually this should become
+/// > `Option<Callable>` once callable proxies are supported.
+pub(crate) fn is_callable(argument: impl TryInto<Function>) -> Option<Function> {
```

Comment:
I think the name should be `to_callable`, right?

Reply:
I thought I'd keep the spec name.

Reply:
The reasoning for keeping the name is that I don't want to have confusion over spec text saying `IsCallable(foo)` and our code then doing `to_callable(foo)` (or maybe rather `try_into_callable`? `to_X` usually suggests a conversion or a clone, I think). While the difference is small, it might end up confusing in the long run.

Now that is not to say that `is_callable` returning a non-bool isn't confusing... But the repeated `Function::try_from(foo).unwrap()` wasn't very nice either.